File: runme/mcgen/local/datan/exasrc/e3rn.c
#include <stdio.h>
#include "datsrc.h"
#include "grsrc.h"

/* Table of constant values */

static integer c__1 = 1, c__100 = 100;
static double c_d1 = 1.;

main(void)
{
    /* Local variables */
    static integer i, n, iswit, nws;
    static double delx, hist[100], a, t, tau1, tau2, x0;
/* identify program to user */
    printf("%s\n\n", "Program E3RN demonstrates use of RNRADI.");
/* ask for numerical input */
    printf("%s\n", "Enter number of decays N (>>1):");
    printf("%s", "> ");
    scanf("%li", &n);
    printf("%s\n", "Enter fraction A (0. < A < 1.):");
    printf("%s", "> ");
    scanf("%lf", &a);
    printf("%s\n", "Enter TAU1 (>0.):");
    printf("%s", "> ");
    scanf("%lf", &tau1);
    printf("%s\n", "Enter TAU2 (>0.):");
    printf("%s", "> ");
    scanf("%lf", &tau2);
/* ask for type of output */
    printf("%s\n", "Choose:");
    printf("%s\n", "1 - Numerical Output");
    printf("%s\n", "2 - Graphical Output");
    printf("%s", "> ");
    scanf("%li", &iswit);
    if (iswit == 1) {
/* numerical output */
        printf("%s%6.2lf%s%6.2lf%s%6.2lf\n",
               "A = ", a, ", TAU1=", tau1, ", TAU2 = ", tau2);
        printf("%s\n", "Decay time values produced by RNRADI:");
	for (i = 1; i <= n; ++i) {
/* simulate decay times */
	    rnradi_(&a, &tau1, &tau2, &t);
            printf("%12.7lf\n ", t);
	}
    } else if (iswit == 2) {
/* graphical output */
        grnbws_();
        printf("%s\n", "Please, enter number of workstation:");
        printf("%s", "> ");
        scanf("%li", &nws);
        (void)getchar();
/* initiate and fill histogram */
	x0 = 0.;
	delx = max(tau1,tau2) * .03;
	smhsin_(hist, &x0, &delx, &c__100);
	for (i = 1; i <= n; ++i) {
/* simulate decay times */
	    rnradi_(&a, &tau1, &tau2, &t);
	    smhsfl_(hist, &x0, &delx, &c__100, &t, &c_d1);
	}
/* present histogram */
	smhsgr_(hist, &x0, &delx, &c__100, "t", &c__1, "N", &c__1, " ", &c__1,
	       &nws);
    }
    return 0;
} /* main */

File: runme/mcgen/local/datan/exasrc/e3rn.f
      PROGRAM E3RN
      IMPLICIT DOUBLE PRECISION(A-H,O-Z)
      PARAMETER(NX=100)
      DIMENSION HIST(NX)
      CHARACTER*1 TX,TY,CAPT
      PARAMETER(TX='t',TY='N',CAPT=' ')
C identify program to user
      WRITE(*,*)' Program E3RN demonstrates use of RNRADI'
      WRITE(*,*)' '
C ask for numerical input
      WRITE(*,*)' Enter number of decays N (>>1)'
      WRITE(*,*)' >'
      READ(*,*)N
      WRITE(*,*)' Enter fraction A (0. < A < 1.)'
      WRITE(*,*)' >'
      READ(*,*)A
      WRITE(*,*)' Enter TAU1 (>0.)'
      WRITE(*,*)' >'
      READ(*,*)TAU1
      WRITE(*,*)' Enter TAU2 (>0.)'
      WRITE(*,*)' >'
      READ(*,*)TAU2
C ask for type of output
      WRITE(*,*)' Choose'
      WRITE(*,*)' 1 - Numerical Output'
      WRITE(*,*)' 2 - Graphical Output'
      WRITE(*,*)' >'
      READ(*,*)ISWIT
      IF(ISWIT.EQ.1)THEN
C numerical output
        WRITE(*,'(3(A,F6.2))')' A = ',A,', TAU1=',TAU1,
     +  ', TAU2 = ',TAU2
        WRITE(*,*)' Decay time values produced by RNRADI:'
        DO 10 I=1,N
C simulate decay times
          CALL RNRADI(A,TAU1,TAU2,T)
          WRITE(*,'(F12.7)')T
10      CONTINUE
      ELSE IF(ISWIT.EQ.2)THEN
C graphical output
        CALL GRNBWS()
        WRITE(*,*)' Please, enter number of workstation:'
        WRITE(*,*)' >'
        READ(*,*)NWS
C initiate and fill histogram
        X0=0.D0
        DELX=0.03D0*MAX(TAU1,TAU2)
        CALL SMHSIN(HIST,X0,DELX,NX)
        DO 20 I=1,N
C simulate decay times
          CALL RNRADI(A,TAU1,TAU2,T)
          CALL SMHSFL(HIST,X0,DELX,NX,T,1.D0)
20      CONTINUE
C present histogram
        CALL SMHSGR(HIST,X0,DELX,NX,TX,1,TY,1,CAPT,1,NWS)
      END IF
      END
